plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/scripts/ScriptEditorPropertyPage.java

```diff
@@ -0,0 +1,164 @@
+        } catch (CoreException e) {
+            log.debug(e.getMessage(), e);
+            return false;            
+        }
+    }
+    
+    /**
+     * Sets value to Disable SQL Editor services property
+     */
+    public static void setDisableEditorServicesProp(IFile file, boolean value) throws CoreException {
+        file.setPersistentProperty(DISABLE_EDITOR_SERVICES_PROP_NAME, Boolean.toString(value));
+        notifyAssociatedServices(file, value);
+    }
+    
+    @Override
+    protected void performDefaults() {
+        super.performDefaults();
+        chkDisableEditorSvcs.setSelection(false);
+    }
+    
+    @Override
+    public boolean performOk() {
+        try {
+            setDisableEditorServicesProp(getCurrentFile(), chkDisableEditorSvcs.getSelection());
+        } catch (CoreException e) {
+            log.debug(e.getMessage(), e);
+            return false;
+        }
+        return true;
+    }
+    
+    
+    private IFile getCurrentFile() {
+        return super.getElement().getAdapter(IFile.class);
+    }
+    
+    private static void notifyAssociatedServices(IFile file, boolean newServicesEnabled) {
+        Set<DBPPreferenceStore> affectedPrefs = new HashSet<>();
+        for (IWorkbenchWindow window : PlatformUI.getWorkbench().getWorkbenchWindows()) {
+            for (IWorkbenchPage page : window.getPages()) {
+                for (IEditorReference editorRef : page.getEditorReferences()) {
+                    IEditorPart editor = editorRef.getEditor(false);
+                    if (editor instanceof SQLEditorBase) {
+                        SQLEditorBase sqlEditor = (SQLEditorBase) editor;
+                        IFile editorFile = editor.getEditorInput().getAdapter(IFile.class);
+                        if (editorFile.equals(file)) {
+                            affectedPrefs.add(sqlEditor.getActivePreferenceStore());
+                        }
+                    }
+                }
+            }
+        }
+        for (DBPPreferenceStore prefs : affectedPrefs) {
+            notifyPrefs(prefs, newServicesEnabled);
+        }
+
+        PlatformUI.getWorkbench().getService(ICommandService.class).refreshElements(DisableEditorServicesHandler.COMMAND_ID, null);
+    }
+    
+    private static void notifyPrefs(DBPPreferenceStore prefStore, boolean newServicesEnabled) {
+        final boolean foldingEnabled = prefStore.getBoolean(SQLPreferenceConstants.FOLDING_ENABLED);
+        final boolean autoActivationEnabled = prefStore.getBoolean(SQLPreferenceConstants.ENABLE_AUTO_ACTIVATION);
+        final boolean markWordUnderCursorEnabled = prefStore.getBoolean(SQLPreferenceConstants.MARK_OCCURRENCES_UNDER_CURSOR);
+        final boolean markWordForSelectionEnabled = prefStore.getBoolean(SQLPreferenceConstants.MARK_OCCURRENCES_FOR_SELECTION);
+        final boolean oldServicesEnabled = !newServicesEnabled;
+        
+        prefStore.firePropertyChangeEvent(
```

Comment:
What is the point in firing events but not changing actual property values?

Reply:
From my point of view, it's a script property, so it should not override datasource preferences. Because of that it triggers property change event and change only effective value of this properties.